aptos-move/framework/aptos-framework/sources/type_map.move

```diff
@@ -0,0 +1,37 @@
+module aptos_framework::type_map {
+    use aptos_std::simple_map::{Self, SimpleMap};
+    use aptos_std::type_info::{Self, TypeInfo};
+    use std::bcs;
+    use aptos_framework::util;
+
+    struct TypeMap has drop, copy, store {
```

Comment:
This is interesting concept. Should we also come up with a MultiTypeMap or KeyedTypeMap<K, T>? I think that would be perfect for Bo's PropertyMap.

Reply:
I add also now the `Any` type. This can be very useful in many scenarios.

Perhaps the KeyedTypeMap should be presented as `SmallMap<Key, TypeMap>`? I can imagine this works well for @areshand properties. Otherwise you maybe able with the help of the new `Any` to rollup whatever polymorphic model you like.

Reply:
Yea, this looks great. Although it doesn't directly support primitive types. maybe, we could start with wrapping primitive types in some struct to use this typemap.